projects/kit/components/tiles/tiles-reorder.providers.ts

```diff
@@ -0,0 +1,29 @@
+import type {FactoryProvider} from '@angular/core';
```

Comment:
That's quite a lot of tokens, how about we create only one: `TUI_TILE_REORDER` and 2 imported methods `tuiTileSwap` and `tuiTileShift`. Swap will be default value for that token and in reorder we will provide shift one. Otherwise it looks a bit over-engineered.